Firmware/xyzcal.cpp

```diff
@@ -13,8 +13,8 @@
 #define XYZCAL_PINDA_HYST_MAX 100 //250um
 #define XYZCAL_PINDA_HYST_DIF 5   //12.5um
 
-#define ENABLE_FANCHECK_INTERRUPT()  EIMSK |= (1<<7)
-#define DISABLE_FANCHECK_INTERRUPT() EIMSK &= ~(1<<7)
+#define ENABLE_FANCHECK_INTERRUPT() ATOMIC_BLOCK(ATOMIC_RESTORESTATE) EIMSK |= _BV(INT7)
+#define DISABLE_FANCHECK_INTERRUPT() ATOMIC_BLOCK(ATOMIC_RESTORESTATE) EIMSK &= ~_BV(INT7)
```

Comment:
if this expands into a single instruction (hypothetically), we don't have to do the critical section

Reply:
It doesn't. It expands to 3 instructions (IN, OR/AND, OUT). Only IO registers from 0x00 to 0x1F are atomic RMW.